packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -12,8 +12,10 @@ export let initialFocus: boolean = false;
 export let id: string | undefined = undefined;
 export let name: string | undefined = undefined;
 export let error: boolean = false;
+// the headings will be shown in the provided order: 1st heading - 1st function result, 2nd heading - 2nd function result, ...
+export let headings: string[] = [];
 
-export let searchFunction: SearchFunction = async (_s: string) => [];
+export let searchFunctions: SearchFunction[] = [async (_s: string) => []];
```

Comment:
As a user, I'm not very confident when I have to fill several correlated arrays. There is always a risk there are not aligned.

I would prefer to have a single array of a structure `{ heading, searchFunction }`

Reply:
I am wondering if it would not be better if we have a single searchFunction, and it would return `Array<{ value: string, group?: string }>`

Reply:
Moreover, I am wondering, if the `Typeahead` should not have a searchFunctions, and instead have a `onChange()` property, and the parent component should be responsible of doing the dirty work.

Then change items from Array<string> to `Array<{value: string, group?: string }>`.